ir/deserializer.go

```diff
@@ -79,7 +79,7 @@ func DeserializePreamble(irBuf []byte) (Deserializer, int, error) {
 		return nil, int(pos), err
 	}
 
-	if 1 != metadataType {
+	if metadataType != 1 {
```

Comment:
so I assume this means we will not use yoda condition in Go?

Reply:
Right. We'll follow the docs + linters and not use yoda conditions and this/self.